File: Algorithms/p415_Add_Strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int l = max(num1.size(),num2.size());
        string res;
        res.reserve(l);
        int j1 = num1.size()-1;
        int j2 = num2.size()-1;
        int carry = 0;
        for(int i=0;i<l;++i){
            int din1 = j1>=0 ? num1[j1]-'0' : 0;
            int din2 = j2>=0 ? num2[j2]-'0' : 0;
            int dout = din1+din2+carry;
            carry = dout>=10 ? 1 : 0;
            dout = dout>=10 ? dout-10 : dout;
            res += '0'+dout;
            if(j1>=0) --j1;
            if(j2>=0) --j2;
        }
        if(carry==1){
            res += '1';
        }
        reverse(res.begin(),res.end());
        return res;
    }
};

File: Algorithms/p415_Add_Strings/p415_Add_Strings.py
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        l = max(len(num1),len(num2))
        res = []
        j1 = len(num1)-1
        j2 = len(num2)-1
        carry = 0
        ord_0 = ord('0')
        for i in range(0,l):
            din1 = ord(num1[j1])-ord_0 if j1>=0 else 0
            din2 = ord(num2[j2])-ord_0 if j2>=0 else 0
            dout = din1+din2+carry
            carry = 1 if dout>=10 else 0
            dout = dout-10 if dout>=10 else dout
            res += chr(ord_0+dout)
            if j1>=0:
                j1 -= 1
            if j2>=0:
                j2 -= 1
        if carry==1:
            res += '1'
        res.reverse()
        return ''.join(res)
